test/unit/datastore/filestore.test.js

```diff
@@ -120,15 +124,20 @@ describe('FileStore', function() {
           'content-length': '2503'
         });
 
-      const data = await store.upload(file, {
+      const promise = store.upload(file, {
         filename: 'kinvey.png',
         public: true,
         mimeType: 'image/png'
-      });
-      expect(data).toIncludeKey('_data');
+      })
+        .then((data) => {
+          expect(data).toIncludeKey('_data');
+          expect(nock.isDone()).toEqual(true);
+          return data;
+        });
+      return promise.should.be.fulfilled;
     });
 
-    it('should resume a file upload when a 308 status code is received', async function() {
+    it('should resume a file upload when a 308 status code is received', function() {
```

Comment:
The test was passing because we had the necessary requests mocked and the result of whether the file was uploaded to GCS or not is still the file data from Kinvey. Since we still were getting back the file data even though the file was not being sent to GCS the test appeared to be passing. I have added a check to ensure that all mocked responses are sent. If a mocked response, such as the GCS upload response, is not sent it will fail now.